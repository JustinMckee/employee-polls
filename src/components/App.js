import {Routes, Route} from 'react-router-dom';
import {useEffect} from 'react';
import {connect} from 'react-redux';
import {handleInitialData} from '../actions/shared';
import './App.scss';

// Design System
import LoadingBar from 'react-redux-loading-bar';
import Container from '@mui/material/Container';

// My Components
import Navigation from './Navigation';
import Dashboard from './Dashboard';
import Leaderboard from './Leaderboard';
import Create from './Create';
import Login from './Login';
import Question from './Question';
import NotFound from './NotFound';

const App = (props) => {

  useEffect(() => {
    props.dispatch(handleInitialData());
  },[props]);

  return (
    <div className="App">

      <LoadingBar />

      <header className="App-header">
        <Navigation />
      </header>
        {
          !props.authedUser && (
            <Container>
              <Routes>
                <Route path="*" state={props} element={<Login />} />
              </Routes>
            </Container>
          )
        }
        {
          props.authedUser && (
            <Container>
              <Routes>
                <Route path="*" state={props} element={<NotFound />} />
                <Route path="/" state={props} exact element={<Dashboard />} />
                <Route path="/Leaderboard" exact element={<Leaderboard />} />
                <Route path="/questions/:id" element={<Question/>} />
                <Route path="/add" exact element={<Create />} />
                <Route path="/login" element={<Login />} />
              </Routes>
            </Container>
          )
        }
    </div>
  );
}

const mapStateToProps = ({authedUser, users}) => (
  {
    loading: authedUser !== null,
    authedUser
  }
)

export default connect(mapStateToProps)(App);
