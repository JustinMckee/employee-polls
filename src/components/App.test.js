import {render} from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux'
import reducer from '../reducers';
import middleware from '../middleware';
import App from './App';
import { BrowserRouter } from "react-router-dom";
import {createStore} from 'redux';

let store;

describe('App matches snapshot', () => {

    beforeEach(async () => {
        store = await createStore(reducer, middleware);
    });
    
    it('Will match snapshot', async () => {
        
        const {component} = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        await expect(component).toMatchSnapshot();
    });
});