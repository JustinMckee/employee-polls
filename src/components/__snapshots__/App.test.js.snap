// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot Will match snapshot 1`] = `undefined`;
